tsk/img/img_writer.cpp

```diff
@@ -234,6 +248,8 @@ static TSK_RETVAL_ENUM addNewBlock(TSK_IMG_WRITER* writer, TSK_OFF_T addr, char
 
     /* Write the new offset to the BAT */
     if (TSK_OK != seekToOffset(writer, writer->batOffset + 4 * blockNum)) {
+        free(fullBuffer);
+        free(sectorBitmap);
```

Comment:
Do you need to also free completedSectors here?

Reply:
The original code never frees completedSectors, because it is assigned outside
`writer->blockToSectorBitmap[blockNum] = completedSectors;`

I am keeping the original logic. Is this ok?

Reply:
Yeah that's fine - was just making sure it wasn't an oversight.